src/web/routes/application/select-address/select-address.js

```diff
@@ -17,22 +19,44 @@ const buildAddressOption = result => ({
   text: result.ADDRESS
 })
 
-const behaviourForGet = () => async (req, res, next) => {
+const behaviourForGet = () => (req, res, next) => {
   res.locals.addresses = req.session.postcodeLookupResults.map(buildAddressOption)
   // Manual address is further in the flow than select-address, therefore this line is needed to prevent the state machine from redirecting the user back to select-address.
   req.session.nextAllowedStep = '/manual-address'
   next()
 }
 
+const findAddress = (selectedAddress, postcodeLookupResults) => {
+  const address = postcodeLookupResults.find(result => result.ADDRESS === selectedAddress)
+
+  if (!address) {
```

Comment:
`Array.prototype.find` will return `undefined` if an address is not matched. I think this check should be a bit more explicit. There is an `isUndefined` function in the common predicates file.